libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Recognizers/CrossTrainedRecognizerSet.cs

```diff
@@ -87,7 +87,7 @@ public override async Task<RecognizerResult> RecognizeAsync(DialogContext dialog
                 var result = await recognizer.RecognizeAsync(dialogContext, activity, cancellationToken, telemetryProperties, telemetryMetrics).ConfigureAwait(false);
                 result.Properties["id"] = recognizer.Id;
                 return result;
-            }));
+            })).ConfigureAwait(false);
```

Comment:
missing cancellation token... otherwise this is beautiful 😞

Reply:
there is no cancelaltion token for Task.WhenAll() the tasks inside have it.

---
In reply to: [422347949](https://github.com/microsoft/botbuilder-dotnet/pull/3892#discussion_r422347949) [](ancestors = 422347949)